opencog/atoms/core/RewriteLink.h

```diff
@@ -44,86 +44,100 @@ class RewriteLink : public ScopeLink
 protected:
 	RewriteLink(Type, const Handle&);
 
-protected:
 	void init(void);
 
+	/**
+	 * Helper for partial_substitute. Given a mapping from
+	 * variables to values, some of which might be variables
+	 * themselves, this generates a new variable declaration
+	 * with using the new, substituted variables.
+	 */
+	Handle partial_substitute_vardecl(const HandleMap& vm) const;
+	static Handle substitute_vardecl(const Handle& vardecl,
+	                                 const HandleMap& vm);
+
+	/**
+	 * Helper for partial_substitute. Given the variable
+	 * declaration and a mapping from variables to values,
+	 * this performs substitution over all bodies(??),
+	 * consuming ill quotations if necessary.
+	 *
+	 * XXX why does it say "all bodies"? How can there be more
+	 * one body?
```

Comment:
I mean terms being scoped, for instance ImplicationScope have 2 "bodies", the antecedent and the consequent.

Reply:
OK, thanks, I  think I get it now.